lib/event_sourcery/aggregate_root.rb

```diff
@@ -124,8 +124,12 @@ def load_history(events)
 
     def apply_event(event_class, options = {})
       event = event_class.new(**options.merge(aggregate_id: id))
-      mutate_state_from(event)
-      @changes << event
+      if event.valid?
```

Comment:
I'd say it's EventSourcery's job to ensure that only valid Events are persisted to the event store so I'd vote for moving the `valid?` check before sinking the event i.e. only validate on write and not read. This validation check could live in the repository but then clients that directly write events without aggregates would miss out on the behaviour so it's likely better to move it to the event sink call itself.

As mentioned elsewhere the split between ES-postgres and ES is rough. It would be great to have this validation check in a general event sink class in ES and then each persistence driver (PG, Memory, Mysql) wouldn't need to worry about it.